### ISSUE TITLE
plugin:typescript/recommended is not compatible with eslint-plugin-vue

### ISSUE THREAD

**Issue by ChristianStornowski** [bradzacher/eslint-plugin-typescript#272]
**Repro**

https://codesandbox.io/s/rj5q83k7lo

**Expected Result**

```
/home/cstornowski/Downloads/src/App.vue
  2:17  error  Disallow self-closing on HTML void elements (<img/>)  vue/html-self-closing
  2:34  error  Attribute "src" should be on a new line               vue/max-attributes-per-line

/home/cstornowski/Downloads/src/components/HelloWorld.vue
   6:39  error  Attribute "target" should be on a new line           vue/max-attributes-per-line
   7:45  error  Attribute "target" should be on a new line           vue/max-attributes-per-line
   9:42  error  Attribute "target" should be on a new line           vue/max-attributes-per-line
  11:47  error  Attribute "target" should be on a new line           vue/max-attributes-per-line
  12:7   error  Disallow self-closing on HTML void elements (<br/>)  vue/html-self-closing
  14:61  error  Attribute "target" should be on a new line           vue/max-attributes-per-line
  22:44  error  Attribute "target" should be on a new line           vue/max-attributes-per-line
  24:44  error  Attribute "target" should be on a new line           vue/max-attributes-per-line
  26:48  error  Attribute "target" should be on a new line           vue/max-attributes-per-line
  29:56  error  Attribute "target" should be on a new line           vue/max-attributes-per-line
```

**Actual Result**

```
/home/cstornowski/Downloads/src/App.vue
  2:7  error  Parsing error: '>' expected

/home/cstornowski/Downloads/src/components/HelloWorld.vue
  2:7  error  Parsing error: '>' expected

/home/cstornowski/Downloads/src/main.ts
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/no-duplicate-attributes
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/no-template-key
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/no-textarea-mustache
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/no-unused-vars
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/require-component-is
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/require-v-for-key
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/valid-v-bind
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/valid-v-cloak
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/valid-v-else-if
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/valid-v-else
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/valid-v-for
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/valid-v-html
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/valid-v-if
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/valid-v-model
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/valid-v-on
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/valid-v-once
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/valid-v-pre
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/valid-v-show
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/valid-v-text
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/attribute-hyphenation
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/html-end-tags
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/html-indent
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/html-self-closing
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/max-attributes-per-line
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/mustache-interpolation-spacing
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/v-bind-style
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/v-on-style
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/attributes-order
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/html-quotes
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/no-confusing-v-for-v-if
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/this-in-template
   1:1  error  Use the latest vue-eslint-parser. See also https://github.com/vuejs/eslint-plugin-vue#what-is-the-use-the-latest-vue-eslint-parser-error  vue/no-multi-spaces
  10:1  error  Expected indentation of 4 spaces but found 2                                                                                              typescript/indent
  11:1  error  Expected indentation of 4 spaces but found 2                                                                                              typescript/indent
  12:1  error  Expected indentation of 4 spaces but found 2                                                                                              typescript/indent
```

**Additional Info**
```
"plugin:typescript/recommended"
```
uses:
```
"parser": "eslint-plugin-typescript/parser",
```

see: https://eslint.vuejs.org/user-guide/#what-is-the-use-the-latest-vue-eslint-parser-error

**Versions**

| package                    | version |
| -------------------------- | ------- |
| `eslint-plugin-typescript` | `1.0.0-rc.1` |
| `typescript`               | `3.1.6` |

**Comment by armano2** [comment]
you have to override it:

```json
  "parser": "vue-eslint-parser",
  "parserOptions": {
      "parser": "eslint-plugin-typescript/parser"
  }
```

**Comment by bradzacher** [comment]
based on what they have written in their docs, it's a chicken and egg scenario due to a weakness in the eslint config.

Their recommended sets `"parser": "vue-eslint-parser"`, and our recommended sets `"parser": "eslint-plugin-typescript/parser"`.

Their docs say that they require you to move our parser to `parserOptions`, but it looks like it could just as easily be done the other way round too.

I didn't even know that eslint could handle two parsers at the same time.

**Comment by armano2** [comment]
@bradzacher no its not eslint.. vue-eslint-parser is using additional parser to parse parts of script from vue files, and fallbacks if its not vue file to `parserOptions.parser`

vue files looks like this:

```vue
<template>
 <div></div>
</template>
<script lang="ts">
@Component
export default class Foo extends Vue {

}
</script>
```

**Comment by bradzacher** [comment]
That makes sense!
I've never used vue, so thanks for explaining.

I'll close this issue then as it's not a bug with our recommended config.
